lesson-12/tests/test_polygon2.py

```diff
@@ -0,0 +1,33 @@
+import unittest
+
+from lesson12 import polygon2
+
+class TestPolygon(unittest.TestCase):
+
+    def test_polygon_object_attribute_initialization(self):
+
+        s1,s2,s3=4,6,6
+        
+        test_poligon = polygon2.Polygon(s1,s2,s3)
+
+        assert hasattr(test_poligon, 's1'), 'Polygon has no Side 1'
+        assert hasattr(test_poligon, 's2'), 'Polygon has no Side 2'
+        assert hasattr(test_poligon, 's3'), 'Polygon has no Side 3'
+
+    
+    def test_sides_sum_equals_perimeter(self):
+
+        pol = polygon2.Polygon(4,4,4)
+
+        test_poligon_perimeter = polygon2.Polygon.perimeter(pol)
```

Comment:
in partea de `act` a testului trebuie sa ne folosim de ce am pregatit mai sus, in cazul asta `pol`. Folosesti `pol` pasandul la metoda perimeter dar o varianta mai usor de inteles ar fi `test_poligon_perimeter = pol.perimeter()`